obs/project.go

```diff
@@ -207,50 +269,71 @@ func (c *Client) GetProjectMetaFile(ctx context.Context, projectName string) (*P
 }
 
 // DeleteProject deletes an existing OBS project
-func (c *Client) DeleteProject(ctx context.Context, project *Project) error {
-	urlPath, err := url.JoinPath(c.APIURL, "source", project.Name)
+func (o *OBS) DeleteProject(ctx context.Context, project *Project) error {
+	urlPath, err := url.JoinPath(o.options.APIURL, "source", project.Name)
 	if err != nil {
 		return fmt.Errorf("deleting obs project: joining url: %w", err)
 	}
 
-	req, err := http.NewRequestWithContext(ctx, http.MethodDelete, urlPath, http.NoBody)
+	resp, err := o.InvokeOBSEndpoint(ctx, http.MethodDelete, urlPath, nil)
 	if err != nil {
+		return fmt.Errorf("deleting obs project: invoking obs endpoint: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		var status Status
+		if err := xml.NewDecoder(resp.Body).Decode(&status); err != nil {
+			return &APIError{
+				HTTPStatusCode: resp.StatusCode,
+				OBSStatusCode:  "",
+				Message:        fmt.Sprintf("deleting obs project: decoding error response %v", err),
+			}
+		}
+
 		return &APIError{
-			HTTPStatusCode: 0,
-			OBSStatusCode:  "",
-			Message:        fmt.Sprintf("deleting obs project: creating request: %v", err),
+			HTTPStatusCode: resp.StatusCode,
+			OBSStatusCode:  status.Code,
+			Message:        status.Summary,
 		}
 	}
 
-	req.SetBasicAuth(c.Username, c.Password)
-	req.Header.Set("Accept", "application/xml; charset=utf-8")
+	return nil
+}
 
-	resp, err := c.Client.Do(req)
+func (o *OBS) ListPackages(ctx context.Context, projectName string) (*Package, error) {
```

Comment:
Let's create a new file for functions related to packages (e.g. `packages.go`)

Reply:
Gotcha!